### PR TITLE
Improve logging of event encryption in `RustCrypto`

### PR DESCRIPTION
**Pull request by richvdh**
It's currently rather hard to see what's going on during event encryption in the rust crypto stack. One of the problems is that you can have multiple encryption operations overlapping.

The main idea here is to include an identifier for the event being encrypted on each log line. (We don't yet have an event ID, so instead use the txn ID.) In order to pass this down through the call hierarchy, I've created the idea of a `LogSpan` (drawing inspiration from [OpenTracing](https://opentracing.io/docs/overview/spans/)).

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by richvdh** [comment]
> (drawing inspiration from [OpenTracing](https://opentracing.io/docs/overview/spans/)).

Maybe we should just be using opentracing itself. That's not a job I particularly want to take on today though.

File: src/rust-crypto/rust-crypto.ts [hunk 1]
```diff
@@ -375,7 +375,7 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
         const encryptor = this.roomEncryptors[room.roomId];
 
         if (encryptor) {
-            encryptor.ensureEncryptionSession(this.globalBlacklistUnverifiedDevices);
+            encryptor.prepareForEncryption(this.globalBlacklistUnverifiedDevices);
```
Comment (andybalaam) [inline src/rust-crypto/rust-crypto.ts:378]: This looks unrelated to the logging change?

**Review by andybalaam — CHANGES_REQUESTED** [review]
One question - looks good otherwise.

File: src/rust-crypto/rust-crypto.ts (hunk 1 above)
Comment (richvdh) [reply src/rust-crypto/rust-crypto.ts:378]: No it's part of the same thing. `ensureEncryptionSession` has gained an extra param and become private. `prepareForEncryption` is a new wrapper for it.

Comment (andybalaam) [reply src/rust-crypto/rust-crypto.ts:378]: Great, thanks.

**Review by andybalaam — APPROVED** [review]

**Comment by richvdh** [comment]
tests failed due to https://github.com/vector-im/element-web/issues/26603. Retrying
